### PR TITLE
Feature 149 join pushdown for exasol dialect

### PR DESCRIPTION
**Pull request by snehlsen**
Join pushdown for the Exasol dialect. (see #149)

### CODE REVIEW

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java [hunk 1]
```diff
@@ -108,12 +108,11 @@ public String visit(final SqlSelectList selectList) throws AdapterException {
...
     public String visit(final SqlColumn column) throws AdapterException {
         String tablePrefix = "";
-        if (this.context.hasMoreThanOneTable()) {
-            if (column.hasTableAlias()) {
-                tablePrefix = this.dialect.applyQuote(column.getTableAlias());
-            } else {
-                tablePrefix = this.dialect.applyQuote(column.getTableName());
-            }
+        if (column.hasTableAlias()) {
+            tablePrefix = this.dialect.applyQuote(column.getTableAlias());
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java]: Contract to one assignment please.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java [hunk 2]
```diff
@@ -108,12 +108,11 @@ public String visit(final SqlSelectList selectList) throws AdapterException {
...
+            tablePrefix = tablePrefix + this.dialect.getTableCatalogAndSchemaSeparator();
+        } else if (column.getTableName() != null && !column.getTableName().isEmpty() ) {
+            tablePrefix = this.dialect.applyQuote(column.getTableName());
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java]: Contract to one assignment please.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/exasol/ExasolSqlDialect.java [hunk 3]
```diff
@@ -63,7 +64,10 @@ public StructureElementSupport supportsJdbcSchemas() {
     @Override
     public Capabilities getCapabilities() {
         final Capabilities.Builder builder = Capabilities.builder();
-        builder.addMain(MainCapability.values());
+        builder.addMain(SELECTLIST_PROJECTION, SELECTLIST_EXPRESSIONS, FILTER_EXPRESSIONS, AGGREGATE_SINGLE_GROUP,
+                AGGREGATE_GROUP_BY_COLUMN, AGGREGATE_GROUP_BY_EXPRESSION, AGGREGATE_GROUP_BY_TUPLE, AGGREGATE_HAVING,
+                ORDER_BY_COLUMN, ORDER_BY_EXPRESSION, LIMIT, LIMIT_WITH_OFFSET,
+                JOIN, JOIN_TYPE_INNER, JOIN_TYPE_LEFT_OUTER, JOIN_TYPE_RIGHT_OUTER, JOIN_TYPE_FULL_OUTER, JOIN_CONDITION_EQUI);
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/exasol/ExasolSqlDialect.java]: :heavy_check_mark:

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 4]
```diff
@@ -1,25 +1,31 @@
...
 import java.io.FileNotFoundException;
 import java.math.BigDecimal;
 import java.net.URI;
-import java.sql.*;
 import java.sql.Date;
-import java.util.*;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: Please contract imports with to "*" if >= 3.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 5]
```diff
@@ -1,25 +1,31 @@
...
-import com.exasol.adapter.dialects.AbstractIntegrationTest;
-import com.exasol.adapter.dialects.IntegrationTestConfigurationCondition;
+import static org.junit.Assert.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 /**
  * Tested with Oracle 12
- *
+ * <p>
  * TODO Add tests for data types TODO Test Expanding of SELECT * if elements of select list require casting
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: Remove TODOs. If relevant --> move to tickets.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 6]
```diff
@@ -268,9 +274,9 @@ void testSelectNumber() throws SQLException {
             matchNextRowDecimal(result, "1234567890123456789012345678901234.6");
         }
         matchSingleRowExplain(String.format(query, EXA_NUMBER_TABLE_JDBC_NUMBER_TO_DECIMAL),
-                "SELECT CAST(\"A\" AS DECIMAL(36,1)) FROM \"" + ORA_NUMBER_TABLE + "\"");
+                "SELECT CAST(\"NUMBER_T\".\"A\" AS DECIMAL(36,1)) FROM \"" + ORA_NUMBER_TABLE + "\"");
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: Use "ORA_NUMBER_TABLE" constant for consistency.

Same for other tables below.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java (hunk 1 above)
Comment (snehlsen) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java]: done

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java (hunk 2 above)
Comment (snehlsen) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java]: done

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java (hunk 4 above)
Comment (snehlsen) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: done

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java (hunk 6 above)
Comment (snehlsen) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: done

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java (hunk 5 above)
Comment (snehlsen) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: done

**Pull request merged**